### PR TITLE
Darkmap

### PR DESCRIPTION
**Pull request by Almightywesley**
new map styles and button

### CODE REVIEW

File: app/src/main/java/nl/hogeschoolrotterdam/projectb/fragment/MapFragment.java [hunk 1]
```diff
@@ -48,8 +48,8 @@
     private EditText mSearchText;
     private LatLng latLng;
     private View tooltip;
-    private ClusterManager<MyItem> clusterManager;
-    private boolean bool = false;
+    private boolean boolMapInitializer = false;
```
Comment (MrTheGood) [inline app/src/main/java/nl/hogeschoolrotterdam/projectb/fragment/MapFragment.java:51]: Thanks for changing this name as I requested previously!

File: app/src/main/java/nl/hogeschoolrotterdam/projectb/fragment/MapFragment.java [hunk 2]
```diff
@@ -176,10 +172,25 @@ public void onInfoWindowClick(Marker marker) {
...
         });
-        //night mode map
-        //MapStyleOptions style = MapStyleOptions.loadRawResourceStyle(getContext(),R.raw.mapstylenight);
-        //googleMap.setMapStyle(style);
-        googleMap.setMapType(GoogleMap.MAP_TYPE_NORMAL);
+        //style mode mode map
+        if (WhibApp.getInstance().getThemeId()== R.style.AppTheme_Dark){
+            MapStyleOptions style = MapStyleOptions.loadRawResourceStyle(getContext(),R.raw.mapstylenight);
+            googleMap.setMapStyle(style);
+        }else if(WhibApp.getInstance().getThemeId()== R.style.AppTheme_Dark_Purple){
```
Comment (MrTheGood) [inline app/src/main/java/nl/hogeschoolrotterdam/projectb/fragment/MapFragment.java]: `ctrl+alt+l` has not been done.

File: app/src/main/java/nl/hogeschoolrotterdam/projectb/fragment/MapFragment.java [hunk 3]
```diff
@@ -176,10 +172,25 @@ public void onInfoWindowClick(Marker marker) {
...
+            googleMap.setMapStyle(style);
+
+        }else if(WhibApp.getInstance().getThemeId()== R.style.AppTheme_Dark_Red){
+            MapStyleOptions style = MapStyleOptions.loadRawResourceStyle(getContext(),R.raw.mapstylered);
+            googleMap.setMapStyle(style);
+
+        }
+        else {
+            MapStyleOptions style = MapStyleOptions.loadRawResourceStyle(getContext(),R.raw.mapstylenormal);
+            googleMap.setMapStyle(style);;
```
Comment (MrTheGood) [inline app/src/main/java/nl/hogeschoolrotterdam/projectb/fragment/MapFragment.java]: There's a double semicolon here.

File: app/src/main/java/nl/hogeschoolrotterdam/projectb/fragment/MapFragment.java [hunk 4]
```diff
@@ -176,10 +172,25 @@ public void onInfoWindowClick(Marker marker) {
...
+        }else if(WhibApp.getInstance().getThemeId()== R.style.AppTheme_Dark_Purple){
+            MapStyleOptions style = MapStyleOptions.loadRawResourceStyle(getContext(),R.raw.mapstyleaubergine);
+            googleMap.setMapStyle(style);
+
+        }else if(WhibApp.getInstance().getThemeId()== R.style.AppTheme_Dark_Red){
+            MapStyleOptions style = MapStyleOptions.loadRawResourceStyle(getContext(),R.raw.mapstylered);
+            googleMap.setMapStyle(style);
+
+        }
+        else {
```
Comment (MrTheGood) [inline app/src/main/java/nl/hogeschoolrotterdam/projectb/fragment/MapFragment.java]: This is missing the new `AppTheme_Dark_Green`

**Review by Hicham010 — APPROVED** [review]
Nice work

File: app/src/main/res/values-nl/strings.xml [hunk 5]
```diff
@@ -9,7 +9,7 @@
 
     <!-- Strings -->
     <string name="str_enter_adress">Adres invullen</string>
-    <string name="str_search">Zoeken…</string>
+    <string name="str_search">Zoek op titel, land</string>
```
Comment (MrTheGood) [inline app/src/main/res/values-nl/strings.xml]: `Zoek op titel, land of beschrijving`

File: app/src/main/java/nl/hogeschoolrotterdam/projectb/fragment/MapFragment.java (hunk 4 above)
Comment (MrTheGood) [reply app/src/main/java/nl/hogeschoolrotterdam/projectb/fragment/MapFragment.java]: Also it does not make sure that new dark themes automatically get some default dark map.

**Review by MrTheGood — CHANGES_REQUESTED** [review]
Here are my first thoughts.
I haven't tested the app yet though, so I might give some other feedback later.

Regarding the outdated screenshots in the settings page: They'll be updated at the end of this sprint.

File: app/src/main/res/layout/fragment_map.xml [hunk 6]
```diff
@@ -43,13 +43,24 @@
...
+    <ImageView
+        android:layout_width="40dp"
+        android:layout_height="40dp"
+        app:layout_constraintEnd_toEndOf="parent"
+        android:layout_margin="10dp"
+        app:layout_constraintTop_toBottomOf="@id/relLayout1"
+        android:scaleType="centerCrop"
+        android:id="@+id/ic_gps"
+        android:src="@drawable/ic_gps"
+        />
```
Comment (MrTheGood) [inline app/src/main/res/layout/fragment_map.xml:64]: After testing the app:
This recenter button should use `android:tint="?colorOnSurface"` to ensure that it is also visible on dark maps.

**Review by MrTheGood — CHANGES_REQUESTED** [review]
I just also tested the app and added one extra comment.

**Comment by MrTheGood** [comment]
Thanks for implementing my feedback.

**Review by MrTheGood — APPROVED** [review]

**Pull request merged**